src-electron/generator/helper-zcl.js

```diff
@@ -169,10 +189,9 @@ async function zcl_structs(options) {
       for (const i of st.items) {
         if (i.isArray) {
           // Found an array. Now let's check if it points to a struct that also contains an array.
-          let sis = await queryZcl.selectAllStructItemsByStructName(
+          let sis = await queryZcl.selectAllStructItemsById(
             this.global.db,
-            i.type,
-            packageIds
+            i.dataTypeReference
```

Comment:
But `i` is an array, not a struct, no?  Is the dataTypeReference here actually correct?  Where is it coming from, exactly?  @brdandu please check whether the query changes above handle this correctly.

Reply:
Reverted changes to fabric-sensitive bits

`i` is the result of `selectAllStructsWithItems`, i.e. a `structItem` (see `db-mapping.js`)

The changes here must be correct, because all tests pass - unless the tests are no good 😅🤣

In which case that should be addressed separately